datafusion/core/src/dataframe/parquet.rs

```diff
@@ -246,4 +249,136 @@ mod tests {
+       
+        for field in schema.fields().iter() {
+            encrypt = encrypt.with_column_key(field.name().as_str(), column_key.clone());
+            decrypt = decrypt.with_column_key(field.name().as_str(), column_key.clone());
+        }
+        
+        let encrypt = encrypt.build().unwrap();
+        let decrypt = decrypt.build().unwrap();
+        
+        let df = test_df.clone();
+        let tmp_dir = TempDir::new()?;
+        let tempfile = tmp_dir.path().join("roundtrip.parquet");
+        let output_path = "file://local/roundtrip.parquet";
+        
+        let num_rows_written: usize = if true {
+            // Write encrypted parquet using ArrowWriter
+            let props = WriterProperties::builder()
+                .with_file_encryption_properties(encrypt)
+                .build();
+
+            let batches = df.collect().await?;
+            write_batches(tempfile.clone(), props, batches).unwrap()
+        } else {
+            // Write encrypted parquet using write_parquet
+            let local = Arc::new(LocalFileSystem::new_with_prefix(&tmp_dir)?);
+            let local_url = Url::parse("file://local").unwrap();
+            let ctx = &test_df.session_state;
+            ctx.runtime_env().register_object_store(&local_url, local);
+            let mut options = TableParquetOptions::default();
+            options.global.file_encryption_properties = Some(encrypt.clone().into());
+            df.write_parquet(
+                output_path,
+                DataFrameWriteOptions::new().with_single_file_output(true),
+                Some(options),
+            )
+                .await?;
+            test_df.count().await?
+        };
+        
+        
+        // Check that file actually used encryption
+        // TODO, I guess arrow-rs does not have a serialized file reader that supports encryption?
+        /*
+        let file = std::fs::File::open(tmp_dir.path().join("test.parquet"))?;
+
+        let reader =
+            parquet::file::serialized_reader::SerializedFileReader::new(file)
+                .unwrap();
+
+        let parquet_metadata = reader.metadata();
+
+        let written_compression =
+            parquet_metadata.row_group(0).column(0).compression();
+
+        assert_eq!(written_compression, parse_compression_string(compression)?);
+        */
+
+        // Read encrypted parquet
+        let mut sc = SessionConfig::new();
+        let fd: ConfigFileDecryptionProperties = decrypt.clone().into();
+        sc.options_mut()
+            .execution
+            .parquet
+            .file_decryption_properties = Some(fd);
+
+        let state = SessionStateBuilder::new().with_config(sc).build();
+        let ctx: SessionContext = SessionContext::new_with_state(state);
+        
+        let encrypted_batches =
+            read_parquet_test_data(tempfile.into_os_string().into_string().unwrap(), &ctx)
+                .await;
```

Comment:
For now, use previous test code of read_parquet_test_data while I debug the write process.